### PR TITLE
Validations and model specs.

### PR DESCRIPTION
**Pull request by amon-cofie**
Changes:

- Added the following validations:
    - For the User model:
        - Name must not be blank.
        - PostsCounter must be an integer greater than or equal to zero.
    - For the Post model:
        - Title must not be blank.
        - Title must not exceed 250 characters.
        - CommentsCounter must be an integer greater than or equal to zero.
        - LikesCounter must be an integer greater than or equal to zero.
        
- Added unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  it 'is not valid if comments_counter is less than 0' do
+    post = Post.new(comments_counter: -1)
+    expect(post).to_not be_valid
+  end
+
+  it 'is not valid if likes_counter is less than 0' do
+    post = Post.new(likes_counter: -1)
+    expect(post).to_not be_valid
+  end
```
Comment (Deepakdanger) [inline spec/models/post_spec.rb]: - [ ] Kindly add some positive test cases also.

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      user.valid?
+      expect(user.errors).to have_key(:name)
+    end
+
+    it 'is invalid with a posts_counter less than 0' do
+      user = User.new(posts_counter: -1)
+      user.valid?
+      expect(user.errors).to have_key(:posts_counter)
+    end
+  end
```
Comment (Deepakdanger) [inline spec/models/user_spec.rb]: - [ ] Kindly add some positive test cases also. For ex
- Create one user and check its validity & its attributes

File: spec/models/comment_spec.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
+RSpec.describe Comment, type: :model do
+  # tests go here
+
```
Comment (Deepakdanger) [inline spec/models/comment_spec.rb:5]: - [ ] Kindly add some of the validation test case here ( +ve & -ve ) cases also

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @amon-cofie , 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/comment_spec.rb (hunk 3 above)
Comment (amon-cofie) [reply spec/models/comment_spec.rb:5]: Please observe this file closely. I believe I do have a few test cases written in the file 😅 
But in any case, I will surely take a good look at the file again. Thank you very much for the great review.

File: spec/models/user_spec.rb (hunk 2 above)
Comment (amon-cofie) [reply spec/models/user_spec.rb]: Thank you for the suggestion. I will make the change

File: spec/models/post_spec.rb (hunk 1 above)
Comment (amon-cofie) [reply spec/models/post_spec.rb]: Thank you very much. Indeed it is a good idea to have both negative and positive test cases in test files. I will make the change immediately

**Review by mirouhml — APPROVED** [review]
# APPROVED 🟢

Hi @amon-cofie, 

Good job making the required changes, your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- Great job setting up the correct linters ✔️
- Highly organized and clean code ✔️ 👨🏻‍💻
- Professional documentation ✔️
- Correct workflow ✔️
- Meaningful commit messages ✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**